floris/tools/floris_interface.py

```diff
@@ -195,6 +195,7 @@ def reinitialize(
         solver_settings: dict | None = None,
         time_series: bool | None = False,
         layout: tuple[list[float], list[float]] | tuple[NDArrayFloat, NDArrayFloat] | None = None,
+        het_map=None,
```

Comment:
It would be ideal to maintain the type hinting here but the heterogeneous map is difficult to express. This is a good candidate for a class defining the heterogeneous map and providing a good interface for interacting with it.